diff --git a/common/private/genfs_contexts b/common/private/genfs_contexts
index 57f7e0e..fe8f51f 100644
--- a/common/private/genfs_contexts
+++ b/common/private/genfs_contexts
@@ -1,5 +1,7 @@
 genfscon sysfs /devices/virtual/timed_output/vibrator u:object_r:sysfs_vibrator:s0
 genfscon sysfs /module/mmcblk/parameters/perdev_minors u:object_r:sysfs_perdev_minors:s0
 
+ifelse(gta3xlwifi_stock_removable_fs, `true', `', `
 genfscon iso9660 / u:object_r:iso9660:s0
 genfscon udf / u:object_r:udf:s0
+')
